libcodechecker/libhandlers/server.py

```diff
@@ -342,6 +392,186 @@ def arg_match(options):
+    prod_status = {}
+    for pd in products:
+        db = database.SQLServer.from_connection_string(pd.connection,
+                                                       RUN_META,
+                                                       migration_root,
+                                                       interactive=False)
+
+        ret = db.connect()
+        s_ver = db.get_schema_version()
+        if s_ver in db_errors:
+            s_ver = None
+        prod_status[pd.endpoint] = (ret, s_ver, package_schema)
+
+    sess.commit()
+    sess.close()
+    engine.dispose()
+
+    return prod_status
+
+
+def __db_status_check(cfg_sql_server, context, product_name=None):
+    """
+    Check and print database statuses for the given product.
+    """
+    if not product_name:
+        return 0
+
+    LOG.debug("Checking database status for " + product_name +
+              " product.")
+
+    prod_statuses = check_product_db_status(cfg_sql_server, context)
+
+    if product_name != 'all':
+        avail = prod_statuses.get(product_name)
+        if not avail:
+            LOG.error("No product was found with this endpoint: " +
+                      str(product_name))
+            return 1
+
+        prod_statuses = {k: v for k, v in prod_statuses.items()
+                         if k == product_name}
+
+    print_prod_status(prod_statuses)
+    return 0
+
+
+def __db_migration(cfg_sql_server, context, product_to_upgrade):
+    """
+    Handle database management.
+    Schema checking and migration.
+    """
+    LOG.info("Migrating schema for " + str(product_to_upgrade))
+    product_name = product_to_upgrade
+
+    prod_statuses = check_product_db_status(cfg_sql_server, context)
+    prod_to_upgrade = []
+
+    if product_name != 'all':
+        avail = prod_statuses.get(product_name)
+        if not avail:
+            LOG.error("No product was found with this endpoint: " +
+                      product_name)
+            return 1
+        prod_to_upgrade.append(product_name)
+    else:
+        prod_to_upgrade = list(prod_statuses.keys())
+
+    migration_root = context.run_migration_root
+
+    for prod in prod_to_upgrade:
+        LOG.info("========================")
+        LOG.info("Upgrading: " + prod)
+        engine = cfg_sql_server.create_engine()
+        config_session = sessionmaker(bind=engine)
+        sess = config_session()
+
+        product = sess.query(ORMProduct).filter(
+                ORMProduct.endpoint == prod).first()
+        db = database.SQLServer.from_connection_string(product.connection,
+                                                       RUN_META,
+                                                       migration_root,
+                                                       interactive=False)
+
+        db_status = db.connect()
+
+        msg = database_status.db_status_msg.get(db_status,
+                                                'Unknown database status')
+
+        nothing_to_do = [DBStatus.OK,
```

Comment:
Do you need this? Instead of listing everything when there is nothing to do this could be just an else branch.